python/cuml/linear_model/base.pyx

```diff
@@ -66,6 +68,21 @@ class LinearPredictMixin:
         Predicts `y` values for `X`.
 
         """
+        coef_cp, n_feat, n_targets, _ = input_to_cupy_array(self.coef_)
+        if 1 < n_targets:
+            # Handle multi-target prediction in Python.
+            X_cp = input_to_cupy_array(
+                X,
+                check_dtype=self.dtype,
+                convert_to_dtype=(self.dtype if convert_dtype else None),
+                check_cols=self.n_cols
+            ).array
+            intercept_cp = input_to_cupy_array(self.intercept_).array
+            preds_cp = X_cp @ coef_cp + intercept_cp
+            preds = input_to_cuml_array(preds_cp).array
```

Comment:
I don't think this explicit conversion is needed, but should actually be handled by the Base class. Maybe @wphicks can weigh in on this?

Reply:
I believe you're correct. The decorator on this should take care of the conversion.

Reply:
I have removed the conversion.